File: _archive/models/paper.py
# Created by DINKIssTyle on 2026.
# Copyright (C) 2026 DINKI'ssTyle. All rights reserved.

"""용지 및 네임태그 레이아웃 데이터 모델"""

from dataclasses import dataclass, field
from typing import Optional


# 용지 프리셋 (이름: (너비mm, 높이mm))
PAPER_PRESETS = {
    "A3": (297.0, 420.0),
    "A4": (210.0, 297.0),
    "A5": (148.0, 210.0),
    "B4": (250.0, 353.0),
    "B5": (176.0, 250.0),
    "Letter": (215.9, 279.4),
    "Legal": (215.9, 355.6),
    "Custom": (210.0, 297.0),
}


@dataclass
class PaperSize:
    """용지 크기"""
    name: str = "A4"
    width_mm: float = 210.0
    height_mm: float = 297.0

    @classmethod
    def from_preset(cls, name: str) -> "PaperSize":
        if name in PAPER_PRESETS:
            w, h = PAPER_PRESETS[name]
            return cls(name=name, width_mm=w, height_mm=h)
        return cls(name="A4", width_mm=210.0, height_mm=297.0)


@dataclass
class TagLayout:
    """네임태그 레이아웃 설정"""
    tag_width_mm: float = 90.0
    tag_height_mm: float = 54.0
    columns: int = 2
    rows: int = 5
    offset_x_mm: float = 15.0
    offset_y_mm: float = 13.5
    gap_x_mm: float = 0.0
    gap_y_mm: float = 0.0

File: models/paper.go
// Created by DINKIssTyle on 2026.
// Copyright (C) 2026 DINKI'ssTyle. All rights reserved.

package models

var PaperPresets = map[string][2]float64{
	"A3":     {297.0, 420.0},
	"A4":     {210.0, 297.0},
	"A5":     {148.0, 210.0},
	"B4":     {250.0, 353.0},
	"B5":     {176.0, 250.0},
	"Letter": {215.9, 279.4},
	"Legal":  {215.9, 355.6},
	"Custom": {210.0, 297.0},
}

// PaperSize 용지 크기
type PaperSize struct {
	Name     string  `json:"name"`
	WidthMM  float64 `json:"width_mm"`
	HeightMM float64 `json:"height_mm"`
}

func DefaultPaperSize() PaperSize {
	return PaperSize{
		Name:     "A4",
		WidthMM:  210.0,
		HeightMM: 297.0,
	}
}

func PaperSizeFromPreset(name string) PaperSize {
	if size, ok := PaperPresets[name]; ok {
		return PaperSize{
			Name:     name,
			WidthMM:  size[0],
			HeightMM: size[1],
		}
	}
	return DefaultPaperSize()
}

// TagLayout 네임태그 레이아웃 설정
type TagLayout struct {
	TagWidthMM  float64 `json:"tag_width_mm"`
	TagHeightMM float64 `json:"tag_height_mm"`
	Columns     int     `json:"columns"`
	Rows        int     `json:"rows"`
	OffsetXMM   float64 `json:"offset_x_mm"`
	OffsetYMM   float64 `json:"offset_y_mm"`
	GapXMM      float64 `json:"gap_x_mm"`
	GapYMM      float64 `json:"gap_y_mm"`
}

func DefaultTagLayout() TagLayout {
	return TagLayout{
		TagWidthMM:  90.0,
		TagHeightMM: 54.0,
		Columns:     2,
		Rows:        5,
		OffsetXMM:   15.0,
		OffsetYMM:   13.5,
		GapXMM:      0.0,
		GapYMM:      0.0,
	}
}
